lib/api/model/events.dart

```diff
@@ -74,6 +80,57 @@ class AlertWordsEvent extends Event {
   Map<String, dynamic> toJson() => _$AlertWordsEventToJson(this);
 }
 
+/// A Zulip event of type `user_settings` with op `update`.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class UserSettingsUpdateEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'user_settings';
+
+  @JsonKey(includeToJson: true)
+  String get op => 'update';
+
+  /// The name of the setting, or null if we don't recognize it.
+  @JsonKey(unknownEnumValue: JsonKey.nullForUndefinedEnumValue)
+  final UserSettingName? property;
+
+  /// The new value, or null if we don't recognize the setting.
+  ///
+  /// The type is `dynamic`, but in handling real events from the API,
+  /// it's safe to cast [value] to the type appropriate for [property]
+  /// (e.g., to say `value as bool` for a boolean setting).
+  /// That's because [UserSettingsUpdateEvent.fromJson] does that check itself,
+  /// and it would have failed if the check didn't pass.
```

Comment:
```suggestion
  /// This will have the type appropriate for [property]; for example,
  /// if the setting is boolean, then `value is bool` will always be true.
  /// This invariant is enforced by [UserSettingsUpdateEvent.fromJson].
```

Bit tighter and perhaps more assertive, which I think is appropriate for describing an invariant.